src/steamcompmgr.cpp

```diff
@@ -586,18 +596,32 @@ static win * find_win( struct wlr_surface *surf )
 }
 
 static void
-release_commit( commit_t &commit )
+destroy_buffer( struct wl_listener *listener, void * )
 {
-	if ( commit.fb_id != 0 )
+	std::lock_guard<std::mutex> lock( wlr_buffer_map_lock );
+	wlr_buffer_map_entry *entry = wl_container_of( listener, entry, listener );
+
+	if ( entry->fb_id != 0 )
 	{
-		drm_drop_fbid( &g_DRM, commit.fb_id );
-		commit.fb_id = 0;
+		drm_drop_fbid( &g_DRM, entry->fb_id );
 	}
 
-	if ( commit.vulkanTex != 0 )
+	if ( entry->vulkanTex != 0 )
+	{
+		vulkan_free_texture( entry->vulkanTex );
```

Comment:
`vulkan_free_texture` is not thread-safe: it accesses `g_mapVulkanTextures` without any mutex.

Reply:
Done, also fixed up the refcounts for thread safety.